app/javascript/components/UpdateCreditCardForm/index.jsx

```diff
@@ -0,0 +1,207 @@
+// instead of having a field, I think to keep it sane I'll have to implement
+// the whole form as a React component?
+
+import React, { useState } from 'react'
+import { loadStripe } from '@stripe/stripe-js'
+import {
+  CardElement,
+  Elements,
+  useStripe,
+  useElements
+} from '@stripe/react-stripe-js'
+import BillingAddressField from './BillingAddressField'
+
+const CreditCardField = ({ onChange }) => {
+  const stripe = useStripe()
+  const elements = useElements()
+
+  return (
+    <>
+      <label
+        htmlFor='stripe-elements'
+        className='block text-sm font-medium text-gray-700'
+      >
+        Credit Card
+      </label>
+      <div className='mt-1'>
+        <CardElement
+          id='stripe-elements'
+          className='block w-full border-gray-300 rounded-md shadow-sm sm:text-sm focus:ring-indigo-500 focus:border-indigo-500'
+          onChange={onChange}
+        />
+      </div>
+    </>
+  )
+}
+
+const SubmitButton = ({ isLoading }) => {
+  const label = isLoading ? (
+    <>
+      <svg
+        className='w-5 h-5 mr-3 -ml-1 text-white animate-spin'
+        xmlns='http://www.w3.org/2000/svg'
+        fill='none'
+        viewBox='0 0 24 24'
+        width='1em'
+        height='1em'
+      >
+        <circle
+          className='opacity-25'
+          cx={12}
+          cy={12}
+          r={10}
+          stroke='currentColor'
+          strokeWidth={4}
+        />
+        <path
+          className='opacity-75'
+          fill='currentColor'
+          d='M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z'
+        />
+      </svg>
+      Saving
+    </>
+  ) : (
+    <>Save</>
+  )
+  return (
+    <button
+      disabled={isLoading}
+      className='inline-flex justify-center px-4 py-2 ml-3 text-sm font-medium text-white bg-indigo-600 border border-transparent rounded-md shadow-sm hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500'
+    >
+      {label}
+    </button>
+  )
+}
+
+const UpdateCreditCardForm = ({
+  action,
+  method,
+  authenticityToken,
+  countryOptions
+}) => {
+  const stripe = useStripe()
+  const elements = useElements()
+  const [isLoading, setIsLoading] = useState(false)
+
+  const handleCardOnChange = event => {
+    // make sure we disable the input in case of error
```

Comment:
This is a `TODO`? It is ok to go without this for now?

Reply:
Yeah, I think we can move on without this for now. I'll put a TODO